fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMutualTest.java

```diff
@@ -524,4 +528,105 @@ void testMutualIndexingWithEmptyFragments() {
         assertAllReadable(indexes);
         validateIndexes(indexes);
     }
+
+    @Test
+    void testSortAndSquash() {
+        List<byte[]> points = new ArrayList<>();
+        points.add(byteEmpty());
+        for (int i = 2; i < 0xff; i += 3) {
+            points.add(byteOf(i));
+        }
+        points.add(byteOf(0xff));
+        List<Range> ranges = new ArrayList<>();
+        for (int i = 0; i < points.size() - 1; i++) {
+            ranges.add(new Range(points.get(i), points.get(i + 1)));
+        }
+        List<Range> partial = new ArrayList<>();
+        for (int i = 1 ; i < ranges.size(); i += 4) {
+            partial.add(ranges.get(i));
+        }
+        // test squash
+        Collections.shuffle(ranges);
+        List<Range> squashed = IndexingMutuallyByRecords.sortAndSquash(ranges);
+        assertEquals(1, squashed.size());
```

Comment:
This seems like a degenerate case. Namely, it seems like it would be better if it ended up with multiple ranges. Perhaps it would be easier to start with 10 distinct ranges, and then take each range and break it into multiple overlapping ranges of different sizes (some just duplicates, some just abutting, some overlapping by a bunch), and then `sortAndSquash` should result in the same list that the test started with.

Reply:
✅